### DISCUSSION TITLE
Reverse for Django Channels

### DISCUSSION THREAD

**Discussion by Archmonger** [django/channels#1912] [Ideas]
Currently, it is a bit disjointed that Django Core has [`reverse` capabilities](https://docs.djangoproject.com/en/4.1/ref/urlresolvers/#reverse) but `channels` does not for websockets.

Perhaps it's worth considering a light wrapper around Django Reverse, that probes `ASGI_APPLICATION["websocket"]` URLs?

There seems to be [an issue from 2018](https://github.com/django/channels/issues/1045) about this. This discussion is to track whether `reverse` is feasible with the latest versions of Channels.

**Comment by carltongibson** [comment]
Hi @Archmonger -- I'm not sure how feasible it is. I think it needs someone to sit down and derive a PoC, but solving the nested ASGI app issue is likely intractable. 🤔

**Reply by Archmonger** [reply]
My immediate thought is optionally allowing a dotted path to a central websocket `urlpatterns` to be defined in `settings.py`.

Would fit Django design patterns, and only be required for certain features such as reverse or resolve.
